deployment/address_book.go

```diff
@@ -82,6 +82,7 @@ type AddressBook interface {
 	AddressesForChain(chain uint64) (map[string]TypeAndVersion, error)
 	// Allows for merging address books (e.g. new deployments with existing ones)
 	Merge(other AddressBook) error
+	Remove(ab AddressBook) error
 }
 
 type AddressBookMap struct {
```

Comment:
This struct uses a map as a storage, but it's not thread-safe.  Do we need to refactor it?

Reply:
yeah I think we need to make this thread-safe